parsl/utils.py

```diff
@@ -24,3 +24,63 @@ def get_version():
         logger.exception("Unable to determine code state")
 
     return version
+
+
+def get_all_checkpoints():
+    """Finds the checkpoints from all last runs.
+
+    Note that checkpoints are incremental, and this helper will not find
+    previous checkpoints from earlier than the most recent run. It probably
+    should be made to do so.
+
+    Returns:
+       - a list suitable for the checkpointFiles parameter of DataFlowKernel
+         constructor
+
+    """
+
+    if(not(os.path.isdir('runinfo/'))):
+        return []
+
+    dirs = sorted(os.listdir('runinfo/'))
+
+    checkpoints = []
+
+    for runid in dirs:
+
+        checkpoint = os.path.abspath('runinfo/{0}/checkpoint'.format(runid))
+
+        if(os.path.isdir(checkpoint)):
+            checkpoints.append(checkpoint)
+
+    return checkpoints
+
+
+def get_last_checkpoint():
+    """Finds the checkpoint from the last run, if one exists.
+
+    Note that checkpoints are incremental, and this helper will not find
+    previous checkpoints from earlier than the most recent run. It probably
+    should be made to do so.
+
+    Returns:
+     - a list suitable for checkpointFiles parameter of DataFlowKernel
+       constructor, with 0 or 1 elements
+
+    """
+
+    if(not(os.path.isdir('runinfo/'))):
+        return []
+
+    dirs = sorted(os.listdir('runinfo/'))
+
+    if(len(dirs) == 0):
+        return []
+
+    last_runid = dirs[-1]
+    last_checkpoint = os.path.abspath('runinfo/{0}/checkpoint'.format(last_runid))
```

Comment:
The updates look good-- I'm still concerned that we are hardcoding the checkpoint directory here though.